server/src/main/java/org/opensearch/index/query/NestedQueryBuilder.java

```diff
@@ -319,9 +319,9 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
         }
 
         BitSetProducer previousParentFilter = context.getParentFilter();
+        context.nestedScope().nextLevel(nestedObjectMapper);
```

Comment:
Just curious why the calls where reordered here? Could `context.nestedScope().nextLevel(...)` behavior change (happy path) because parent filter is not set?

Reply:
`.previousLevel()` call in `finally` block hardly makes sense if `.nextLevel()` throw an exception. I encounter this edge case when test set `index.query.max_nested_depthindex.query.max_nested_depth=0` (btw is it a good min boundary?), to me it means - no nested. Then  `.nextLevel()` throws boundary exception, and `.previousLevel()` hit another exception attempting to pop from empty stack.

Reply:
> .previousLevel() call in finally block hardly makes sense if .nextLevel() throw an exception.

That is not a happy path, right? 

> . I encounter this edge case when test set index.query.max_nested_depthindex.query.max_nested_depth=0 (btw is it a good min boundary?),

Apparently it is not, we probably should at least have a value of 1, what do you think?

Reply:
> That is not a happy path, right?
Right. it's a really bad path.

> should at least have a value of 1, 
Well. I don't have an opinion. Having min 0 let to completely prohibit nested queries. Have no idea who might use it. 

so, put `min=1`?

Reply:
@mkhludnev still have concern about changing the sequence here:

```
context.setParentFilter(parentFilter);
context.nestedScope().nextLevel(nestedObjectMapper);
```

The `nestedScope()` or `nestedScope().nextLevel(nestedObjectMapper)` may somehow rely that `parentFilter` is set, I think it would be safe to keep the sequence unchanged. 

(apologies to unresolving this one, I was doing the last pass and this thing stood out once again).

Reply:
Ok. @reta , just put lines as they were. I have to add one finally block since now `previousLevel()` should be called only after successful completion of `nextLevel()`

Reply:
TY